lib/ts/recipe/session/sessionFunctions.ts

```diff
@@ -87,77 +87,75 @@ export async function getSession(
-                    throw err;
-                }
-                /**
-                 * if it comes here, it means token verification has failed.
-                 * It may be due to:
-                 *  - signing key was updated and this token was signed with new key
-                 *  - access token is actually expired
-                 *  - access token was signed with the older signing key
-                 *
-                 * if access token is actually expired, we don't need to call core and
-                 * just return TRY_REFRESH_TOKEN to the client
-                 *
-                 * if access token creation time is after this signing key was created
-                 * we need to call core as there are chances that the token
-                 * was signed with the updated signing key
-                 *
-                 * if access token creation time is before oldest signing key was created,
-                 * so if foundOlderKey is still false after the loop we just return TRY_REFRESH_TOKEN
-                 */
-                let payload;
-                try {
-                    payload = getPayloadWithoutVerifiying(accessToken);
-                } catch (_) {
-                    throw err;
-                }
-                if (payload === undefined) {
-                    throw err;
-                }
+                payload = getPayloadWithoutVerifiying(accessToken);
+            } catch (_) {
+                throw err;
+            }
+            if (payload === undefined) {
+                throw err;
+            }
 
-                const timeCreated = sanitizeNumberInput(payload.timeCreated);
-                const expiryTime = sanitizeNumberInput(payload.expiryTime);
+            const timeCreated = sanitizeNumberInput(payload.timeCreated);
+            const expiryTime = sanitizeNumberInput(payload.expiryTime);
 
-                if (expiryTime === undefined || expiryTime < Date.now()) {
-                    throw err;
-                }
+            if (expiryTime === undefined || expiryTime < Date.now()) {
+                throw err;
+            }
 
-                if (timeCreated === undefined) {
-                    throw err;
-                }
+            if (timeCreated === undefined) {
+                throw err;
+            }
 
-                // This is kind of like the old signingKeyLastUpdated logic using the creation time
-                // of the oldest key instead of the last update time
-                if (timeCreated >= key.createdAt) {
-                    foundOlderKey = true;
-                    break;
-                }
+            // This is kind of like the old signingKeyLastUpdated logic using the creation time
+            // of the oldest key instead of the last update time
+            if (timeCreated >= key.createdAt) {
+                foundASigningKeyThatIsOlderThanTheAccessToken = true;
+                break;
```

Comment:
Add a comment here explaining the logic behind the if statement.

Reply:
There was already a comment above this, but I expanded it a bit anyway.